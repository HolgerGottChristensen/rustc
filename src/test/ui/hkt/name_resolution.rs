// The %j2 should not be able to be found, because it is not a parameter of the HKT
// check-fail

fn test<I<%J>>(input: I<u32>) {
}

fn main() {
    test::<Option<%J2>>(Some(42)); //~ ERROR could not resolve `%j2`
}
