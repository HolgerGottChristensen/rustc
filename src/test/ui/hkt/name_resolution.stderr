error[E9999]: hkt argument `%J2` could not be found in the definition
  --> $DIR/name_resolution.rs:8:20
   |
LL |     test::<Option<%J2>>(Some(42));
   |                    ^^
   |
note: expected one of the parameters `%J` inside the definition
  --> $DIR/name_resolution.rs:4:9
   |
LL | fn test<I<%J>>(input: I<u32>) {
   |         ^

error: aborting due to previous error

